src/f0/app_solar/source/tmp101.c

```diff
@@ -0,0 +1,261 @@
+/* - SECTION - Driver exported functions.                                    */
+/*===========================================================================*/
+
+/**
+ * @brief   Initializes an instance.
+ *
+ * @param[out] devp     pointer to the @p TMP101Driver object
+ *
+ * @init
+ */
+void tmp101ObjectInit(TMP101Driver *devp) {
+    devp->vmt = &vmt_device;
+
+    devp->config = NULL;
+
+    devp->state = TMP101_STOP;
+}
+
+
+
+
+/**
+ * @brief   Configures and activates TMP101 Complex Driver peripheral.
+ *
+ * @param[in] devp      pointer to the @p TMP101Driver object
+ * @param[in] config    pointer to the @p TMP101Config object
+ *
+ * @api
+ */
+void tmp101Start(TMP101Driver *devp, const TMP101Config *config) {
+//    i2cbuf_t buf;
+
+    osalDbgCheck((devp != NULL) && (config != NULL));
+    osalDbgAssert((devp->state == TMP101_STOP) ||
+            (devp->state == TMP101_READY),
+            "tmp101Start(), invalid state");
+
+    devp->config = config;
+
+    /* Configuring common registers.*/
+
+// 2020-02-24 Note - as of this date no common registers to configure
+//  for TMP101AN sensors.
+
+    devp->state = TMP101_READY;
+
+}
+
+/**
+ * @brief   Deactivates the TMP101 Complex Driver peripheral.
+ *
+ * @param[in] devp       pointer to the @p TMP101Driver object
+ *
+ * @api
+ */
+void tmp101Stop(TMP101Driver *devp) {
+//    i2cbuf_t buf;
+
+    osalDbgCheck(devp != NULL);
+    osalDbgAssert((devp->state == TMP101_STOP) || (devp->state == TMP101_READY),
+            "tmp101Stop(), invalid state");
+
+//    if (devp->state == TMP101_READY) {
+//    }
+    devp->state = TMP101_STOP;
+}
+
+
+
+// 2021-02-07 - first draft read temperature routine
+
+bool read_tmp101an_temperature_v1(TMP101Driver *devp, unsigned int option)
+{
+// 2021-02-07 - these two byte arrays will be passed to a development
+//  routine, to confirm we can talk with the TMP101AN sensors of the
+//  solar board.  When that's working we'll move these and the dev'
+//  code over to the TMP101AN driver source file.  That will be the
+//  first step to honor and conform to the object oritented device
+//  driver encapsulation that's already expressed in the Oresat firmware
+//  project - TMH
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    uint8_t buffer_tx[2] = {0};
+    uint8_t buffer_rx[5] = {0};
+
+    msg_t i2c_result = MSG_OK;
+    i2cflags_t i2c_flags = 0;
+
+    uint8_t sensor_addr = devp->config->saddr;
+
+    if (option == 2)
+    {
+        // option parameter not yet used - TMH
+    }
+
+// Values chosen per TMP101AN datasheet:
+    buffer_tx[0] = TMP101_REG_TEMPERATURE_READING;
+    buffer_tx[1] = 0;
+
+// Clear first two bytes of receive buffer, as temperature reading has 12 bits spanning two bytes:
+    buffer_rx[0] = 0;
+    buffer_rx[1] = 0;
+
+
+#ifdef DIAG_ON__TMP101AN
+    chprintf((BaseSequentialStream *) &SD2, "reading temperature sensor %02X . . .\r\n", sensor_addr);
```

Comment:
Drivers should not have any chprintf related stuff in it

Reply:
Good catch, apologies!  Removed.